packages/edit-site/src/components/page-pages/index.js

```diff
@@ -19,19 +19,23 @@ import { useState, useEffect, useMemo } from '@wordpress/element';
 import Page from '../page';
 import Link from '../routes/link';
 import PageActions from '../page-actions';
-import { DataViews, PAGE_SIZE_VALUES } from '../dataviews';
+import { DataViews } from '../dataviews';
 
 const EMPTY_ARRAY = [];
 const EMPTY_OBJECT = {};
 
 export default function PagePages() {
-	const [ reset, setResetQuery ] = useState( ( v ) => ! v );
-	const [ globalFilter, setGlobalFilter ] = useState( '' );
-	const [ paginationInfo, setPaginationInfo ] = useState();
-	const [ { pageIndex, pageSize }, setPagination ] = useState( {
-		pageIndex: 0,
-		pageSize: PAGE_SIZE_VALUES[ 0 ],
+	const [ view, setView ] = useState( {
+		type: 'table',
+		search: '',
+		page: 0,
+		perPage: 5,
+		sort: {
+			column: 'date',
+			direction: 'desc',
+		},
```

Comment:
Right now, the visible columns is an internal property, I think we may want to add this to this object later.

Reply:
Yes. Fields visibility should be tied to a view and in the future also probably handle some more meta like visibility in mobile.

Reply:
There are also some field metadata related to UI that could also be part of the view. For example, we currently set [maxWidth](https://github.com/WordPress/gutenberg/blob/trunk/packages/edit-site/src/components/page-pages/index.js#L97) via the fields (and there could be other [widths](https://tanstack.com/table/v8/docs/guide/migrating#update-column-definitions) according to the only docs I could find).

I'm thinking of a situation where the views are customized per user/role/etc. or even updated real-time in collaborative editing. In that scenario, the width of the columns seems like an important UI detail to maintain in the view _entity_ – so it can be updated/synchronized/etc.